### PR TITLE
DeploymentManager: API Spec, examples - private preview

### PR DESCRIPTION
**Pull request by deveshguha**
Adding support for DeploymentManager API. This is going to be in private preview.

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by hovsepm** [comment]
@deveshguha please fix semantic validation errors https://travis-ci.org/Azure/azure-rest-api-specs/jobs/423166678

**Comment by hovsepm** [comment]
@deveshguha once your feature will be ready for merge please reach out to me directly.

File: specification/deploymentmanager/resource-manager/readme.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+---
+# Code Generation
+
+
+## Swagger to SDK
+This section describes what SDK should be generated by the automatic system.
+This is not used by Autorest itself.
+
+## C#
```
Comment (hovsepm) [inline specification/deploymentmanager/resource-manager/readme.md]: please add all the languages that your SDK should eventually be published in

Comment (deveshguha) [reply specification/deploymentmanager/resource-manager/readme.md]: @hovsepm synced with @salameer on excluding this from auto generating SDKs. Is it still required to have the required declarations in this file?

**Comment by deveshguha** [comment]
Would you add @salameer to the PR as well? He suggested that I add him here.

File: specification/deploymentmanager/resource-manager/readme.md (hunk 1 above)
Comment (hovsepm) [reply specification/deploymentmanager/resource-manager/readme.md]: @deveshguha it is not required to generate private SDK but is required for this PR to be merged eventually.

Comment (deveshguha) [reply specification/deploymentmanager/resource-manager/readme.md]: Got it. Updated the file referencing other services to add sdks and relevant settings.

File: specification/deploymentmanager/resource-manager/Microsoft.DeploymentManager/preview/2018-09-01-preview/deploymentmanager.json [hunk 2]
```diff
@@ -0,0 +1,1868 @@
...
+        },
+        {
+          "$ref": "#/parameters/ApiVersionParameter"
+        }
+      ],
+      "put": {
+        "tags": [
+          "ServiceTopologies"
+        ],
+        "operationId": "ServiceTopologies_Create",
```
Comment (hovsepm) [inline specification/deploymentmanager/resource-manager/Microsoft.DeploymentManager/preview/2018-09-01-preview/deploymentmanager.json]: >Create [](start = 42, length = 6)

if this method is also used in update scenario then operation ID should end with "CreateOrUpdate`

File: specification/deploymentmanager/resource-manager/Microsoft.DeploymentManager/preview/2018-09-01-preview/deploymentmanager.json [hunk 3]
```diff
@@ -0,0 +1,1868 @@
...
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DeploymentManager/serviceTopologies/{serviceTopologyName}/services/{serviceName}": {
+      "put": {
+        "tags": [
+          "Services"
+        ],
+        "operationId": "Services_Create",
```
Comment (hovsepm) [inline specification/deploymentmanager/resource-manager/Microsoft.DeploymentManager/preview/2018-09-01-preview/deploymentmanager.json]: >Create [](start = 33, length = 6)

per description this operation should end with `CreateOrUpdate`

File: specification/deploymentmanager/resource-manager/Microsoft.DeploymentManager/preview/2018-09-01-preview/deploymentmanager.json [hunk 4]
```diff
@@ -0,0 +1,1868 @@
...
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DeploymentManager/serviceTopologies/{serviceTopologyName}/services/{serviceName}/serviceUnits/{serviceUnitName}": {
+      "put": {
+        "tags": [
+          "ServiceUnits"
+        ],
+        "operationId": "ServiceUnits_Create",
```
Comment (hovsepm) [inline specification/deploymentmanager/resource-manager/Microsoft.DeploymentManager/preview/2018-09-01-preview/deploymentmanager.json]: >Create [](start = 37, length = 6)

`CreateOrUpdate`

File: specification/deploymentmanager/resource-manager/Microsoft.DeploymentManager/preview/2018-09-01-preview/deploymentmanager.json [hunk 5]
```diff
@@ -0,0 +1,1868 @@
...
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DeploymentManager/steps/{stepName}": {
+      "put": {
+        "tags": [
+          "Steps"
+        ],
+        "operationId": "Steps_Create",
```
Comment (hovsepm) [inline specification/deploymentmanager/resource-manager/Microsoft.DeploymentManager/preview/2018-09-01-preview/deploymentmanager.json]: >Create [](start = 30, length = 6)

CreateOrUpdate ?

File: specification/deploymentmanager/resource-manager/Microsoft.DeploymentManager/preview/2018-09-01-preview/deploymentmanager.json [hunk 6]
```diff
@@ -0,0 +1,1868 @@
...
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DeploymentManager/rollouts/{rolloutName}": {
+      "put": {
+        "tags": [
+          "Rollouts"
+        ],
+        "operationId": "Rollouts_Create",
```
Comment (hovsepm) [inline specification/deploymentmanager/resource-manager/Microsoft.DeploymentManager/preview/2018-09-01-preview/deploymentmanager.json]: >Rollouts_Create [](start = 24, length = 15)

is this operation updatable?

File: specification/deploymentmanager/resource-manager/Microsoft.DeploymentManager/preview/2018-09-01-preview/deploymentmanager.json [hunk 7]
```diff
@@ -0,0 +1,1868 @@
...
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DeploymentManager/artifactSources/{artifactSourceName}": {
+      "put": {
+        "tags": [
+          "ArtifactSources"
+        ],
+        "operationId": "ArtifactSources_Create",
```
Comment (hovsepm) [inline specification/deploymentmanager/resource-manager/Microsoft.DeploymentManager/preview/2018-09-01-preview/deploymentmanager.json]: >Create [](start = 40, length = 6)

updatable?

**Comment by hovsepm** [comment]
@deveshguha does your service at the current stage support Listing objects by subscription and resource group? if so - please consider adding them as well (can be done as a separate PR)

**Comment by deveshguha** [comment]
> @deveshguha does your service at the current stage support Listing objects by subscription and resource group? if so - please consider adding them as well (can be done as a separate PR)

Not at this time. We will be following up on this later when we decide to do a broader release.

**Review by hovsepm — APPROVED** [review]

**Pull request merged**